src/app/SharpRaven/Data/QueueRequester.cs

```diff
@@ -0,0 +1,44 @@
+﻿#if !NET40 && !NET35
+using System;
+using System.Collections.Concurrent;
+using System.Threading.Tasks;
+
+namespace SharpRaven.Data
+{
+    public class QueueRequester : IRequester
+    {
+        private readonly IRequester requester;
+        private readonly IProducerConsumerCollection<IRequester> queue;
+
+
+        public QueueRequester(IRequester requester, IProducerConsumerCollection<IRequester> queue)
+        {
+            this.requester = requester ?? throw new ArgumentNullException(nameof(requester));
+            this.queue = queue ?? throw new ArgumentNullException(nameof(queue));
+        }
+
+
+        private string EnqueueRequester()
+        {
+            var eventId = Guid.NewGuid().ToString("n");
+            this.requester.UseEventId(eventId);
+
+            return this.queue.TryAdd(this.requester) 
+                ? eventId 
+                // If it didn't get queued, it won't be sent.
```

Comment:
Might be worth commenting that the item will always be added with `ConcurrentQueue` which is what we're defaulting to

Reply:
True.